front/app/utils/FormattedCurrency/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import useAppConfiguration from 'hooks/useAppConfiguration';
+import { isNilOrError } from 'utils/helperUtils';
+
+// i18n
+import { injectIntl } from 'utils/cl-intl';
+import { InjectedIntlProps, FormattedNumber } from 'react-intl';
+import messages from './messages';
+
+interface Props {
+  value: number;
+}
+
+const FormattedCurrency = ({
+  value,
+  intl: { formatMessage, formatNumber },
+}: Props & InjectedIntlProps) => {
+  const appConfiguration = useAppConfiguration();
+
+  if (!isNilOrError(appConfiguration)) {
+    const currency = appConfiguration.data.attributes.settings.core.currency;
+
+    // custom implementations for custom currencies
+    // see appConfiguration.ts for all currencies
+    if (currency === 'TOK') {
+      return (
+        <>
+          {formatMessage(
```

Comment:
Had to implement like this, because the usual way we deal with this (search for `plural` in messages.ts files) doesn't work with the formatNumber output. Perhaps there's a way to deal with it but thinking pragmatically with the tight deadline.